src/main/java/com/google/flourbot/execution/MacroExecutionModuleImplementation.java

```diff
@@ -146,19 +146,33 @@ public ChatResponse execute(String userEmail, String message, String threadId) t
       case SHEET_READ_ROW:
         int row = ((SheetReadRowAction) action).getRow();
         List<String> rowData = cloudSheet.readRow(row);
-        chatResponse = ChatResponse.createChatResponseWithList(rowData, actionType, documentUrl);
+        if (((SheetReadRowAction) action).getSelectRandomEntry()) {
+          chatResponse = new ChatResponse(selectRandomFromList(rowData), actionType, documentUrl);
```

Comment:
If you need to call a different constructor if the list has a single element use rowData.length() == 1 as the condition instead.

Reply:
I changed it to return a list with a single entry, so the same constructor can be used